### PR TITLE
Move account suspension-related methods to concern

### PR DESCRIPTION
**Pull request by mjankowski**
I've been looking for areas to reduce LOC and/or class size in the "god classes" in the app -- User, Status, Account, MediaAttachment, etc -- and though of this as one possibility.

We could potentially do similar for sensitize/silencing in this class, but I wanted to start with just one idea, will add others if we like this.

### CODE REVIEW

File: app/models/concerns/account/suspensions.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  alias unavailable? suspended?
+
+  def suspended_permanently?
+    suspended? && deletion_request.nil?
+  end
+
+  def suspended_temporarily?
+    suspended? && deletion_request.present?
+  end
+  alias permanently_unavailable? suspended_permanently?
```
Comment (ThisIsMissEm) [inline app/models/concerns/account/suspensions.rb:23]: This should probably be up on line 18/19? (if the alias's are meant to be with the method declaration they alias?)

**Review by ThisIsMissEm — COMMENTED** [review]
I'm liking this pattern, just one small issue noted.

File: app/models/concerns/account/suspensions.rb (hunk 1 above)
Comment (mjankowski) [reply app/models/concerns/account/suspensions.rb]: Probably should, I just copied it as-is. Updated to move up.

**Review by oneiros — APPROVED** [review]

**Comment by mjankowski** [comment]
With this merged, would love to get thoughts on any/all of - https://github.com/mastodon/mastodon/pull/28866 / https://github.com/mastodon/mastodon/pull/28351 / https://github.com/mastodon/mastodon/pull/31270 - which are all similar in goals/changes.

**Comment by mjankowski** [comment]
This one has merged, but several other PRs which are basically the exact same thing but around a different feature/aspect of some model are still sitting open.

Is this just a matter of someone having time to get to it ... or do we for some reason like the pattern for suspension only but not for any of the rest?

These are all almost a year old, I'd love to get them reviewed or close them.

**Pull request merged**